lib/mongoid/clients/options.rb

```diff
@@ -40,13 +40,23 @@ def mongo_client
       end
 
       def persistence_context
-        PersistenceContext.get(self) ||
-            PersistenceContext.get(self.class) ||
-            PersistenceContext.new(self.class)
+        # TODO: Add feature flag
```

Comment:
Good catch, this is a leftover. Since this goes to 9.0, there is no requirement to hide this behind a flag. And I do not think that the existing behaviour is desireable.